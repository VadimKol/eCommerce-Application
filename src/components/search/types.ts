import type { CatalogAction } from '@/common/types';

export interface SearchProps {
  searchField: React.RefObject<HTMLInputElement>;
  dispatch: React.Dispatch<CatalogAction>;
}
