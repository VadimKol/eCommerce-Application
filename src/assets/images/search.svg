<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0px" y="0px" viewBox="0 0 42 52.5" enable-background="new 0 0 42 42" xml:space="preserve"><path d="M33.851,32.439l-7.401-7.402c3.339-4.053,3.117-10.076-0.671-13.865c-1.951-1.951-4.546-3.025-7.305-3.025  s-5.354,1.074-7.305,3.025c-4.027,4.029-4.026,10.582,0,14.609c1.951,1.951,4.546,3.025,7.305,3.025c2.423,0,4.718-0.83,6.563-2.354  l7.399,7.4L33.851,32.439z M12.583,24.367c-3.247-3.248-3.247-8.533,0-11.781c1.573-1.572,3.666-2.439,5.891-2.439  s4.317,0.867,5.891,2.439c3.247,3.248,3.247,8.533,0,11.781c-1.573,1.572-3.666,2.439-5.891,2.439S14.156,25.939,12.583,24.367z"/><text x="0" y="57" fill="#000000" font-size="5px" font-weight="bold" font-family="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif">Created by AVAM</text><text x="0" y="62" fill="#000000" font-size="5px" font-weight="bold" font-family="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif">from the Noun Project</text></svg>