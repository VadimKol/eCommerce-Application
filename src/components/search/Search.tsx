import styles from './styles.module.scss';
import type { SearchProps } from './types';

export function Search({ searchField, dispatch }: SearchProps): JSX.Element {
  return (
    <form
      className={styles.search_form}
      onSubmit={(e) => {
        e.preventDefault();
        if (typeof searchField.current?.value === 'string') {
          dispatch({ type: 'SET_SEARCH', search: searchField.current?.value.trim() });
        }
      }}
    >
      <div className={styles.search_box}>
        <input id="search" className={styles.search} type="text" placeholder="Search..." ref={searchField} />
        <button type="submit" className={styles.search_button} aria-label="Search-button" />
      </div>
    </form>
  );
}
