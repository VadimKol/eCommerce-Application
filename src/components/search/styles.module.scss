.search_form {
  display: flex;
  justify-content: flex-end;

  & .search_box {
    position: relative;

    @media screen and (width <= 590px) {
      width: 100%;
    }
  }

  & .search {
    box-sizing: border-box;
    width: 250px;
    height: 30px;
    padding: 0 30px 0 5px;
    font-family: var(--primary-font);
    font-size: 1em;
    color: var(--primary-color);
    border: 1px solid var(--primary-color);
    border-radius: 6px;

    &:focus {
      border: 1px solid #000000;
    }

    &:focus-visible {
      outline: none;
    }

    @media screen and (width <= 590px) {
      width: 100%;
    }
  }

  & .search_button {
    position: absolute;
    top: 6px;
    right: 6px;
    width: 20px;
    height: 20px;
    padding: 0;
    background-color: transparent;
    background-image: url("/src/assets/images/search.svg");
    background-size: cover;
    border: none;


    @media (hover: hover) {
      &:hover {
        cursor: pointer;
      }
    }
  }

  @media screen and (width <= 590px) {
    width: 100%;
  }
}
