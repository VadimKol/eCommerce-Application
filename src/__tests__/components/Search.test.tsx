import { render } from '@testing-library/react';
import React from 'react';

import { Search } from '@/components/search/Search';

it('renders correctly', () => {
  const someDispatch = (): void => {};
  const searchField: React.RefObject<HTMLInputElement> = { current: document.createElement('input') };
  const useRef = jest.fn().mockImplementation(() => searchField);
  jest.spyOn(React, 'useRef').mockImplementation(useRef);
  const { container } = render(<Search searchField={searchField} dispatch={someDispatch} />);

  expect(container).toMatchSnapshot();
});
