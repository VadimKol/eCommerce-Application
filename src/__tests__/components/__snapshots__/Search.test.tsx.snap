// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    class="search_form"
  >
    <div
      class="search_box"
    >
      <input
        class="search"
        id="search"
        placeholder="Search..."
        type="text"
      />
      <button
        aria-label="Search-button"
        class="search_button"
        type="submit"
      />
    </div>
  </form>
</div>
`;
